kafka/adminapi_test.go

```diff
@@ -58,22 +65,32 @@ func TestAdminAPIWithDefaultValue(t *testing.T) {
 				ReplicationFactor: -1,
 			},
 		})
-	if err != nil {
-		adminClient.Close()
-		t.Fatalf("Failed to create topics %v\n", err)
+	if err != errCheck {
+		t.Errorf("CreateTopics() failed to return expected result")
+		if !a.isClosed {
+			a.Close()
+			t.Fatalf("Failed to create topics %v\n", err)
+		}
+	}
+	if !a.isClosed {
+		t.Logf("Succeed to create topic %v\n", res)
 	}
-	t.Logf("Succeed to create topic %v\n", res)
 
 	ctx, cancel = context.WithTimeout(context.Background(), expDuration)
 	defer cancel()
-	res, err = adminClient.DeleteTopics(ctx, []string{topic})
-	if err != nil {
-		adminClient.Close()
-		t.Fatalf("Failed to delete topic %v, err: %v", topic, err)
+	res, err = a.DeleteTopics(ctx, []string{topic})
+	if err != errCheck {
+		t.Errorf("DeleteTopics() failed to return expected result")
```

Comment:
Print both err and errCheck (in this case, and other cases where expected error is being used)